src/metal.cs

```diff
@@ -2050,5 +2399,242 @@ interface MTLStageInputOutputDescriptor : NSCopying
 		void Reset ();
 	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLType
+	{
+		[Export ("dataType")]
+		MTLDataType DataType { get; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	interface MTLPointerType
+	{
+		[Export ("elementType")]
+		MTLDataType ElementType { get; }
+
+		[Export ("access")]
+		MTLArgumentAccess Access { get; }
+
+		[Export ("alignment")]
+		nuint Alignment { get; }
+
+		[Export ("dataSize")]
+		nuint DataSize { get; }
+
+		[Export ("elementIsArgumentBuffer")]
+		bool ElementIsArgumentBuffer { get; }
+
+		[NullAllowed, Export ("elementStructType")]
+		MTLStructType ElementStructType { get; }
+
+		[NullAllowed, Export ("elementArrayType")]
+		MTLArrayType ElementArrayType { get; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(MTLType))]
+	interface MTLTextureReferenceType
+	{
+		[Export ("textureDataType")]
+		MTLDataType TextureDataType { get; }
+
+		[Export ("textureType")]
+		MTLTextureType TextureType { get; }
+
+		[Export ("access")]
+		MTLArgumentAccess Access { get; }
+
+		[Export ("isDepthTexture")]
+		bool IsDepthTexture { get; }
+	}
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	interface IMTLCaptureScope { }
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface MTLCaptureScope
+	{
+		[Abstract]
+		[Export ("beginScope")]
+		void BeginScope ();
+
+		[Abstract]
+		[Export ("endScope")]
+		void EndScope ();
+
+		[Abstract]
+		[NullAllowed, Export ("label")]
+		string Label { get; set; }
+
+		[Abstract]
+		[Export ("device")]
+		IMTLDevice Device { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("commandQueue")]
+		IMTLCommandQueue CommandQueue { get; }
+	}
+
+
+	[Mac (10,13, onlyOn64: true), iOS (11,0), TV (11,0), NoWatch]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface MTLCaptureManager
+	{
+		[Static]
+		[Export ("sharedCaptureManager")]
+		MTLCaptureManager SharedCaptureManager { get; }
+
+		[Export ("newCaptureScopeWithDevice:")]
+		IMTLCaptureScope CreateNewCaptureScope (IMTLDevice device);
+
+		[Export ("newCaptureScopeWithCommandQueue:")]
+		IMTLCaptureScope CreateNewCaptureScope (IMTLCommandQueue commandQueue);
+
+		[Export ("startCaptureWithDevice:")]
+		void StartCapture (IMTLDevice device);
+
+		[Export ("startCaptureWithCommandQueue:")]
+		void StartCapture (IMTLCommandQueue commandQueue);
+
+		[Export ("startCaptureWithScope:")]
+		void StartCapture (IMTLCaptureScope captureScope);
+
+		[Export ("stopCapture")]
+		void StopCapture ();
+
+		[NullAllowed, Export ("defaultCaptureScope", ArgumentSemantic.Strong)]
+		IMTLCaptureScope DefaultCaptureScope { get; set; }
+
+		[Export ("isCapturing")]
+		bool IsCapturing { get; }
+	}
+
+	[Mac (10, 13, onlyOn64: true), NoiOS, NoTV, NoWatch]
+	enum MTLDeviceNotificationName {
```

Comment:
since that's only exposed in a delegate handler (where enums can't be used) this can be turned into a static type, exposing `NSString` properties.
To make this better add an `[Advice]` on `MTLDeviceNotificationHandler` with a text like:
"The 'NSString' argument will match a property of 'MTLDeviceNotificationHandler'."